backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -47,48 +46,46 @@
     @Inject
     MeterRegistry meterRegistry;
 
-    private Counter getUsersCalls;
-    private Timer getUsersProcessTime;
-    private Counter getGroupUsersCalls;
-    private Timer getGroupUsersProcessTime;
+    private Counter failuresCounter;
 
     @PostConstruct
     public void initCounters() {
-        getUsersCalls = meterRegistry.counter("rbac-recipient-users-provider.get-users.calls");
-        getUsersProcessTime = meterRegistry.timer("rbac-recipient-users-provider.get-users.process-time");
-        getGroupUsersCalls = meterRegistry.counter("rbac-recipient-users-provider.get-group-users.calls");
-        getGroupUsersProcessTime = meterRegistry.timer("rbac-recipient-users-provider.get-group-users.process-time");
+        failuresCounter = meterRegistry.counter("rbac.failures");
     }
 
     @CacheResult(cacheName = "rbac-recipient-users-provider-get-users")
     public Uni<List<User>> getUsers(String accountId, boolean adminsOnly) {
-        getUsersCalls.increment();
-        LocalDateTime start = LocalDateTime.now();
+        Timer.Sample getUsersTotalTimer = Timer.start(meterRegistry);
         return getWithPagination(
-            page -> retryOnError(
-                    rbacServiceToService
-                            .getUsers(accountId, adminsOnly, page * rbacElementsPerPage, rbacElementsPerPage)
-            )
+            page -> {
+                Timer.Sample getUsersPageTimer = Timer.start(meterRegistry);
+                return retryOnError(
+                        rbacServiceToService.getUsers(accountId, adminsOnly, page * rbacElementsPerPage, rbacElementsPerPage)
+                )
+                .onItem().invoke(() -> getUsersPageTimer.stop(meterRegistry.timer("rbac.get-users.page", "accountId", accountId)));
+            }
         )
-        .onItem().invoke(() -> getUsersProcessTime.record(Duration.between(start, LocalDateTime.now())))
+        .onItem().invoke(() -> getUsersTotalTimer.stop(meterRegistry.timer("rbac.get-users.total", "accountId", accountId)))
```

Comment:
I'm wondering if it would make sense to add the total elements that were fetched.

Reply:
Done.